### ISSUE TITLE
plotDataHeatmap() plots interpretation

### ISSUE THREAD

**Issue by merlevede** [bioFAM/MOFA#45]
Hello,

I am not sure I understood properly what shows the plotDataHeatmap().
In the documentation, we find: 

> Function to visualize the loadings for a given set of factors in a given view. 
> This is useful to visualize the overall pattern of the weights but not to individually characterise the factors. 

Let us assume I have 2 layers, gene expression and methylation both at gene level. Moreover, from the plotVarianceExplained(), I see that the factor 1 is represented by both layers. 
When looking at the heatmap obtained from plotDataHeatmap(MOFAobject, view = "mRNA", factor=1), I see features that are enriched in factor 1 in each patient.
Are the genes that are enriched the result of the integration of both layers or only genes enriched in gene expression? In particular, if instead of methylation at gene level I have it at CpG level, it would be impossible to show CpGs in the gene expression view.

To summarize, I have 2 questions:
- From which layer(s) come from the features in this plot? I am confused because the matrices W contain the features * the components and the matrix Z contains the components * nb of samples. So, how can we have the features and the samples? Is it one of the reconstructed Y matrices?
- Are the values represented in the heatmap the input values for these particular features in the layer in question (gene expression in this case)?

Thank you,
Jane

**Comment by rargelaguet** [comment]
Hi Jane,
sorry for not replying. Did you figure this out?

**Comment by merlevede** [comment]
Hello Ricard,

More or less.
Could you please tell confirm the following:
The values plotted in plotDataHeatmap() for layer i are : Wi * Z. Is it correct?

**Comment by rargelaguet** [comment]
No, the values plotted are not the predictions (Wi*Z) but the observations measurements (Y).
In MOFA v2 we included the option to plot the predictions, rather than the data: `plot_data_heatmap(denoise=T)`

**Comment by rargelaguet** [comment]
P.S. If you have missing values in Y, and do `plotDataHeatmap(imputed=TRUE)` then the NAs will be indeed filled with the predictions

**Comment by merlevede** [comment]
Ok, I see.
Thank you
